### PR TITLE
Removed Non Null Check

### PR DESCRIPTION
**Pull request by aryan-more**
Removed unnecessary ArgumentError.checkNotNull as dart became null safe starting from Dart 2.12

### CODE REVIEW

File: sdk/lib/io/data_transformer.dart [hunk 1]
```diff
@@ -526,20 +510,17 @@ class _FilterSink extends ByteConversionSink {
         _sink.add(out);
       }
     } catch (e) {
-      // TODO(kevmoo): not sure why this isn't a try/finally
```
Comment (lrhn) [inline sdk/lib/io/data_transformer.dart:529]: Or why the sink isn't closed if it throws.
Looks weird. Consider keeping the comment, until someone investigates.

Maybe it assumes that only the sink `add` can throw, but that's optimistic, and why is the filter operation inside the `try` then?

**Review by lrhn — APPROVED** [review]

**Comment by devoncarew** [comment]
@lrhn - you'll need to review the associated CL.

**Comment by lrhn** [comment]
The file has some overly long lines, it probably needs to be `dart format`ted again.

File: sdk/lib/io/data_transformer.dart (hunk 1 above)
Comment (aryan-more) [reply sdk/lib/io/data_transformer.dart:529]: Ok

**Comment by aryan-more** [comment]
> The file has some overly long lines, it probably needs to be `dart format`ted again.

My vscode format settings messed up formatting.

**Comment by aryan-more** [comment]
@lrhn why did PR got closed?

**Comment by lrhn** [comment]
Success!
The associated [Gerrit CL](https://dart-review.googlesource.com/c/sdk/+/308421), created from this GitHub PR by the copybara script, was accepted and landed.

**Pull request closed without merging**